framework/src/utils/MooseMeshUtils.C

```diff
@@ -92,27 +92,40 @@ getBoundaryIDs(const libMesh::MeshBase & mesh,
       break;
     }
 
-    libMesh::boundary_id_type id;
-    std::istringstream ss(boundary_name[i]);
-
-    if (!(ss >> id) || !ss.eof())
+    // Check if boundary name duplicate exists and infer id from duplicate
+    bool has_duplicate = false;
+    for (unsigned int j = 0; j < i; j++)
     {
-      /**
-       * If the conversion from a name to a number fails, that means that this must be a named
-       * boundary.  We will look in the complete map for this sideset and create a new name/ID pair
-       * if requested.
-       */
-      if (generate_unknown &&
-          !MooseUtils::doesMapContainValue(sideset_map, std::string(boundary_name[i])) &&
-          !MooseUtils::doesMapContainValue(nodeset_map, std::string(boundary_name[i])))
-        id = ++max_boundary_id;
-      else
-        id = boundary_info.get_id_by_name(boundary_name[i]);
+      if (boundary_name[j] == boundary_name[i])
```

Comment:
a duplicate? So like the user called the function with a repeated entry in their vector?

Reply:
If the input vector contains a boundary ID that is not in the input mesh, then the line `id = ++max_boundary_id;` is hit regardless of if that boundary ID is repeated within the `boundary_name` vector. The correct logic would be to determine whether a duplicate exists in the `boundary_name` vector and then assign the same newly generated value to the duplicates, instead of assigning two different new values to two boundary names that are shared.

Reply:
seems like a bug fix to me